packages/next/kotti-field-toggle/components/ToggleBox.vue

```diff
@@ -0,0 +1,159 @@
+<template>
+	<div class="kt-field-toggle-box">
+		<div class="kt-field-toggle-box__inner">
+			<svg
+				class="kt-field-toggle-box__check"
+				fill="none"
+				height="8"
+				viewBox="0 0 9 8"
+				width="9"
+				xmlns="http://www.w3.org/2000/svg"
+			>
+				<path
+					d="M1 3.92761L2.92242 6L8 1"
+					stroke-linecap="round"
+					stroke-width="2"
+				/>
+			</svg>
+		</div>
+	</div>
+</template>
+
+<script lang="ts">
+import { defineComponent } from '@vue/composition-api'
+
+export default defineComponent({
+	name: 'ToggleBox',
+})
+</script>
+
+<style lang="scss">
```

Comment:
styles in a separate file?

Reply:
This is the styles-only component of the `checkbox__box` (we only stylize the "box" itself here) 

---

1. `ToggleInner` includes `v-if` / `v-else` on `ToggleBox` or `ToggleSwitch`. It is the logic wrapper for both the switch and checkbox because they basically internally are both `<input type='checkbox' .../>`
2. `ToggleBox.vue` &  `ToggleSwitch.vue` are the style components, one works on `checkbox__box`, other works for the stadium part of the switch
3. `KtFieldToggle` is the field itself, it wraps `<ToggleInner />` 
4. to extend 3, there is also `KtFieldToggleGroup` which basically wraps multiple `ToggleInner` (`v-for` depending on how many options)

sorry to be writing all this when u may have possibly seen it but just making sure we r on the same page

---

1. no point (in our opinion though) unless u see smth specific ofc to move out styles from `ToggleBox` anywhere else since that's its sole purpose as a file.
2. if u mean in general have separate style sheets for folders with multiple things, i think we only do it gets hugee (example `KtFieldSelect`s and `KtFieldDate`s) -- for obvious reasons - the styles are at least 200+ lines there 
but other than that - this component specially is realistically JUST to style 
3. If u meant smth entirely different pls say

Reply:
yeah makes sense, I was reasoning from it getting towards huge, but script is small, so this is basically about style.  Also, it was more to raise the point that styles are sometimes separated and wondered about the logic, so thanks for explaining it.  All of this is not a blocker either for me.